pkg/controller/install/errors.go

```diff
@@ -38,11 +38,11 @@ func IsErrorUnrecoverable(err error) bool {
 	if err == nil {
 		return false
 	}
-	_, ok := unrecoverableErrors[reasonForError(err)]
+	_, ok := unrecoverableErrors[ReasonForError(err)]
 	return ok
 }
 
-func reasonForError(err error) string {
+func ReasonForError(err error) string {
```

Comment:
Not related to the PR, but this function is confusing. Since we own the `StrategyError` we should be consistent in using either the error type or a pointer. Checking for both seems a little off.